### PR TITLE
Always print ltac variable type in coercion error message

### LINKED ISSUE #8209: Confusing error message when passing an ident to cbv
#### Version

8.8.1

#### Description of the problem

```coq
Definition foo := Set.
Goal foo.
  let f := ident:(foo) in cbv [f]. (* Ltac variable f is bound to foo which cannot be coerced to an evaluable reference. *)
```

This error message is confusing.  It should say "Ltac variable f is bound to the fresh identifier foo, which cannot be coerced to an evaluable reference."  (Because "foo" certainly can be coerced.)

See also #8208

### PR DESCRIPTION
**Pull request by SkySkimmer**
Close #8209

### CODE REVIEW

File: plugins/ltac/taccoerce.ml [hunk 1]
```diff
@@ -407,14 +407,16 @@ let pr_argument_type arg =
 (** TODO: unify printing of generic Ltac values in case of coercion failure. *)
 
 (* Displays a value *)
-let pr_value env v =
+let pr_value ?(and_type=false) env v =
```
Comment (ppedrot) [inline plugins/ltac/taccoerce.ml:410]: This is dead code, isn't it?

Comment (SkySkimmer) [reply plugins/ltac/taccoerce.ml:410]: WDYM?

Comment (ppedrot) [reply plugins/ltac/taccoerce.ml:410]: Where is and_type used?

File: plugins/ltac/taccoerce.ml [hunk 2]
```diff
@@ -447,7 +449,7 @@ exception CoercionError of Id.t * (Environ.env * Evd.evar_map) option * Val.t *
 let () = CErrors.register_handler begin function
 | CoercionError (id, env, v, s) ->
   Some (str "Ltac variable " ++ Id.print id ++
-   strbrk " is bound to" ++ spc () ++ pr_value env v ++ spc () ++
+   strbrk " is bound to" ++ spc () ++ pr_value ~and_type:true env v ++ spc () ++
```
Comment (SkySkimmer) [inline plugins/ltac/taccoerce.ml:452]: here

Comment (ppedrot) [reply plugins/ltac/taccoerce.ml:452]: But this is not used in pr_value? I don't understand why you add this argument.

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**